Drop Hash `except` core extension

Earlopain opened the pull request:
### Motivation / Background

Rails 8.0 will be Ruby 3.1 or greater only
`except` natively got added in Ruby 3.0 so this is dead code now.

https://docs.ruby-lang.org/en/3.1/Hash.html#method-i-except

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [ ] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

skipkayhil commented on guides/source/active_support_core_extensions.md:
```diff
@@ -2881,27 +2881,25 @@ NOTE: Defined in `active_support/core_ext/object/deep_dup.rb`.
+#### `except!`
 
-The method [`except`][Hash#except] returns a hash with the keys in the argument list removed, if present:
+The method [`except!`][Hash#except!] is identical to the build-in `except` method but removes keys in place, returning `self`.
```
```suggestion
The method [`except!`][Hash#except!] is identical to the built-in `except` method but removes keys in place, returning `self`.
```

Earlopain committed:
Drop Hash `except` core extension

Rails 8.0 will be Ruby 3.1 or greater only
`except` natively got added in Ruby 3.0 so this is dead code now

jhawthorn commented:
Thank you!

vipulnsward commented:
This needs a changelog entry?

Earlopain commented:
There is no behaviour change. On Ruby 3.0 and up, `{}.respond_to?(:except)` is always true since it got added into ruby itself. Because of the condition method definition it was now also always using the native method (for performance reasons I assume)
